server/game/cards/23-AHaH/TheWardenOfTheWest.js

```diff
@@ -14,8 +17,7 @@ class TheWardenOfTheWest extends DrawCard {
         
         this.reaction({
             when: {
-                // TODO: Implement player-aggregate so it only looks at cards being discarded from an individuals hands rather than aggregating all cards being discarded at once
-                'onCardDiscarded:aggregate': event => this.getNumberToDraw(event) > 0 && this.parent.isParticipating()
+                'onCardDiscarded:controller-aggregate': event => this.getNumberToDraw(event) > 0 && this.parent.isParticipating()
```

Comment:
Since we're aggregating by controller, you'd need to check that the controller is an opponent here (your own discards during the challenge would get aggregated)

Reply:
This should be happening within the `getNumberToDraw(event)` function; it filters all events by opponents, and then returns the total number being discarded from hand or deck. If it's your own hand, this would always return 0 (and thus fail this condition)